### ISSUE TITLE
deb package install fails with  unable to create temporary keystore

### ISSUE THREAD

**Issue by LeeDr** [elastic/elasticsearch#41726]
**Elasticsearch version** (`bin/elasticsearch --version`): Version: 8.0.0-SNAPSHOT, Build: default/deb/73bfdc4/2019-04-30T08:55:06.107343Z, JVM: 12.0.1

**Plugins installed**: default distribution

**JVM version** (`java -version`):  JVM: 12.0.1 (packed in deb package)

**OS version** (`uname -a` if on a Unix-like system): Ubuntu 16.04;
Linux vagrant 4.4.0-31-generic #50-Ubuntu SMP Wed Jul 13 00:07:12 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**: Elasticsearch fails to start and logs this in /var/log/syslog;
`unable to create temporary keystore at [/etc/elasticsearch/elasticsearch.keystore.tmp], please check filesystem permissions`

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. get master (8.0.0) snapshot build of elasticsearch deb package with jdk (with or without jdk probably doesn't matter)
 2. install the package `dpkg -i elasticsearch-8.0.0-SNAPSHOT-amd64.deb`
 3. set JAVA_HOME pointing to a suitable JDK.  I used;
`export JAVA_HOME=/usr/share/elasticsearch/jdk`
4. `service elasticsearch start`
5. `tail -n100 /var/log/syslog`

**Provide logs (if relevant)**:
```
May  1 20:57:25 vagrant systemd[1]: Started Elasticsearch.
May  1 20:57:26 vagrant elasticsearch[6191]: OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
May  1 20:57:29 vagrant elasticsearch[6191]: Exception in thread "main" org.elasticsearch.bootstrap.BootstrapException: org.elasticsearch.cli.UserException: unable to create temporary keystore at [/etc/elasticsearch/elasticsearch.keystore.tmp], please check filesystem permissions
May  1 20:57:29 vagrant elasticsearch[6191]: Likely root cause: java.nio.file.AccessDeniedException: /etc/elasticsearch/elasticsearch.keystore.tmp
May  1 20:57:29 vagrant elasticsearch[6191]: #011at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:90)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at java.base/java.nio.file.Files.newOutputStream(Files.java:222)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.apache.lucene.store.FSDirectory$FSIndexOutput.<init>(FSDirectory.java:411)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.apache.lucene.store.FSDirectory$FSIndexOutput.<init>(FSDirectory.java:407)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.apache.lucene.store.FSDirectory.createOutput(FSDirectory.java:255)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.common.settings.KeyStoreWrapper.save(KeyStoreWrapper.java:467)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.bootstrap.Bootstrap.loadSecureSettings(Bootstrap.java:240)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:303)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.cli.Command.main(Command.java:90)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:115)
May  1 20:57:29 vagrant elasticsearch[6191]: #011at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92)
May  1 20:57:29 vagrant elasticsearch[6191]: Refer to the log for complete error details.
May  1 20:57:29 vagrant systemd[1]: elasticsearch.service: Main process exited, code=exited, status=1/FAILURE
May  1 20:57:29 vagrant systemd[1]: elasticsearch.service: Unit entered failed state.
May  1 20:57:29 vagrant systemd[1]: elasticsearch.service: Failed with result 'exit-code'.
```

**Comment by LeeDr** [comment]
I should note that I tried `chmod 777 /etc/elasticsearch` and then elasticsearch started without errors (but I'm sure that's not the right fix!).

**Comment by jasontedor** [comment]
That is most definitely not the right fix and incredibly dangerous!

**Comment by LeeDr** [comment]
Looks like I hit this same error on elasticsearch-oss-8.0.0-SNAPSHOT-x86_64.rpm

```
18:24:24 Exception in thread "main" org.elasticsearch.bootstrap.BootstrapException: org.elasticsearch.cli.UserException: unable to create temporary keystore at [/etc/elasticsearch/elasticsearch.keystore.tmp], please check filesystem permissions
18:24:24 Likely root cause: java.nio.file.AccessDeniedException: /etc/elasticsearch/elasticsearch.keystore.tmp
```

**Comment by cnd** [comment]
why is that closed...

**Comment by tvernum** [comment]
This was fixed in https://github.com/elastic/elasticsearch/pull/41755

**Comment by hrvoj3e** [comment]
Fix on version 6.8.0 (from https://github.com/elastic/elasticsearch/pull/41755/commits/63013fc8307a201eb7ab02dcb3a1d7cbbdc190b7)
```shell
/usr/share/elasticsearch/bin/elasticsearch-keystore create
```
